openmc/examples.py

```diff
@@ -0,0 +1,619 @@
+
+    """
+    model = openmc.model.Model()
+
+    # Define materials.
+    fuel = openmc.Material(name='Fuel')
+    fuel.set_density('g/cm3', 10.29769)
+    fuel.add_nuclide("U234", 4.4843e-6)
+    fuel.add_nuclide("U235", 5.5815e-4)
+    fuel.add_nuclide("U238", 2.2408e-2)
+    fuel.add_nuclide("O16", 4.5829e-2)
+
+    clad = openmc.Material(name='Cladding')
+    clad.set_density('g/cm3', 6.55)
+    clad.add_nuclide("Zr90", 2.1827e-2)
+    clad.add_nuclide("Zr91", 4.7600e-3)
+    clad.add_nuclide("Zr92", 7.2758e-3)
+    clad.add_nuclide("Zr94", 7.3734e-3)
+    clad.add_nuclide("Zr96", 1.1879e-3)
+
+    hot_water = openmc.Material(name='Hot borated water')
+    hot_water.set_density('g/cm3', 0.740582)
+    hot_water.add_nuclide("H1", 4.9457e-2)
+    hot_water.add_nuclide("O16", 2.4672e-2)
+    hot_water.add_nuclide("B10", 8.0042e-6)
+    hot_water.add_nuclide("B11", 3.2218e-5)
+    hot_water.add_s_alpha_beta('c_H_in_H2O')
+
+    # Define the materials file.
+    model.materials = (fuel, clad, hot_water)
+
+    # Instantiate ZCylinder surfaces
+    pitch = 1.26
+    fuel_or = openmc.ZCylinder(x0=0, y0=0, R=0.39218, name='Fuel OR')
+    clad_or = openmc.ZCylinder(x0=0, y0=0, R=0.45720, name='Clad OR')
+    left = openmc.XPlane(x0=-pitch/2, name='left', boundary_type='reflective')
+    right = openmc.XPlane(x0=pitch/2, name='right', boundary_type='reflective')
+    bottom = openmc.YPlane(y0=-pitch/2, name='bottom',
+                           boundary_type='reflective')
+    top = openmc.YPlane(y0=pitch/2, name='top', boundary_type='reflective')
+
+    # Instantiate Cells
+    fuel_pin = openmc.Cell(name='cell 1', fill=fuel)
+    cladding = openmc.Cell(name='cell 3', fill=clad)
+    water = openmc.Cell(name='cell 2', fill=hot_water)
+
+    # Use surface half-spaces to define regions
+    fuel_pin.region = -fuel_or
+    cladding.region = +fuel_or & -clad_or
+    water.region = +clad_or & +left & -right & +bottom & -top
+
+    # Create root universe
+    model.geometry.root_universe = openmc.Universe(0, name='root universe')
+    model.geometry.root_universe.add_cells([fuel_pin, cladding, water])
+
+    model.settings.batches = 10
+    model.settings.inactive = 5
+    model.settings.particles = 100
+    model.settings.source = openmc.Source(space=openmc.stats.Box(
+        [-pitch/2, -pitch/2, -1], [pitch/2, pitch/2, 1], only_fissionable=True))
+
+    plot = openmc.Plot.from_geometry(model.geometry)
+    plot.pixels = (300, 300)
+    plot.color_by = 'material'
+    model.plots.append(plot)
+
+    return model
+
+
+def pwr_core():
+    """Create a PWR full-core model.
+
+    This model is the OECD/NEA Monte Carlo Performance benchmark which is a
+    grossly simplified pressurized water reactor (PWR) with 241 fuel
+    assemblies.
+
+    Returns
+    -------
+    model : openmc.model.Model
+        Full-core PWR model
+
+    """
+    model = openmc.model.Model()
+
+    # Define materials.
+    fuel = openmc.Material(name='Fuel', material_id=1)
```

Comment:
Since this PR makes the examples public-facing, I tend to think that the IDs should be auto-generated. I can already see a new user asking what the rhyme and reason is for the IDs. Unless you think you can clearly articulate that rhyme-n-reason, I'd suggest preempting those questions by just removing the manually assigned IDs.